src/main/cc/wfa/virtual_people/core/model/utils/distributed_consistent_hashing_test.cc

```diff
@@ -0,0 +1,250 @@
+  auto distribution = absl::make_unique<std::vector<DistributionChoice>>();
+  distribution->emplace_back(0, 0);
+  distribution->emplace_back(1, 0);
+  auto hashing =
+      DistributedConsistentHashing::Build(std::move(distribution));
+  EXPECT_EQ(hashing.status().code(), absl::StatusCode::kInvalidArgument);
+}
+
+TEST(DistributedConsistentHashingTest, TestNegativeProbability) {
+  auto distribution = absl::make_unique<std::vector<DistributionChoice>>();
+  distribution->emplace_back(0, -1);
+  auto hashing =
+      DistributedConsistentHashing::Build(std::move(distribution));
+  EXPECT_EQ(hashing.status().code(), absl::StatusCode::kInvalidArgument);
+}
+
+TEST(DistributedConsistentHashingTest, TestOutputDistribution) {
+  // Distribution:
+  // choice_id probability
+  // 0         0.4
+  // 1         0.2
+  // 2         0.2
+  // 3         0.2
+  auto distribution = absl::make_unique<std::vector<DistributionChoice>>();
+  distribution->emplace_back(0, 0.4);
+  distribution->emplace_back(1, 0.2);
+  distribution->emplace_back(2, 0.2);
+  distribution->emplace_back(3, 0.2);
+  ASSERT_OK_AND_ASSIGN(
+      std::unique_ptr<DistributedConsistentHashing> hashing,
+      DistributedConsistentHashing::Build(std::move(distribution)));
+  absl::flat_hash_map<int32_t, int> ouptut_counts = {
+      {0, 0},
+      {1, 0},
+      {2, 0},
+      {3, 0}};
+  for (int seed = 0; seed < kSeedNumber; ++seed) {
+    int32_t output = hashing->Hash(std::to_string(seed));
+    // The output should always be one of [0, 1, 2, 3].
+    EXPECT_TRUE(ouptut_counts.find(output) != ouptut_counts.end());
+    ++ouptut_counts[output];
+  }
+  // Absolute error more than 2% is very unlikely.
+  EXPECT_NEAR(
+      static_cast<double>(ouptut_counts[0]) / static_cast<double>(kSeedNumber),
+      0.4, 0.02);
+  EXPECT_NEAR(
+      static_cast<double>(ouptut_counts[1]) / static_cast<double>(kSeedNumber),
+      0.2, 0.02);
+  EXPECT_NEAR(
+      static_cast<double>(ouptut_counts[2]) / static_cast<double>(kSeedNumber),
+      0.2, 0.02);
+  EXPECT_NEAR(
+      static_cast<double>(ouptut_counts[3]) / static_cast<double>(kSeedNumber),
+      0.2, 0.02);
+}
+
+TEST(DistributedConsistentHashingTest, TestNormalize) {
+  // Distribution:
+  // choice_id probability_before_normalized normalized_probability
+  // 0         0.8                           0.4
+  // 1         0.4                           0.2
+  // 2         0.4                           0.2
+  // 3         0.4                           0.2
+  auto distribution = absl::make_unique<std::vector<DistributionChoice>>();
+  distribution->emplace_back(0, 0.8);
+  distribution->emplace_back(1, 0.4);
+  distribution->emplace_back(2, 0.4);
+  distribution->emplace_back(3, 0.4);
+  ASSERT_OK_AND_ASSIGN(
+      std::unique_ptr<DistributedConsistentHashing> hashing,
+      DistributedConsistentHashing::Build(std::move(distribution)));
+  absl::flat_hash_map<int32_t, int> ouptut_counts = {
```

Comment:
This should be 'output_counts' instead of 'ouptut_counts'

Reply:
Done.